synth.js

```diff
@@ -1,157 +1,175 @@
-
-
-
-/* 
- *  Some presets 
- */
-
-var kick = {
-  type: 'sine',
-  frequency: 100,
-  duration: .3,
-  ramp: {
-    to: 70
-  }
-};
-
-var snare = {
-  type: 'square',
-  frequency: 100,
-  duration: .05,
-};
-
-var beep_a = {
-  type: 'triangle',
-  frequency: 440,
-  duration: .2,
-  filter: {
-    frequency: 1000,
-    type: 'highpass',
-    gain: 25
-  }
-};
-
-var beep_d = {
-  type: 'triangle',
-  frequency: 587.33,
-  duration: .2,
-  filter: {
-    frequency: 1000,
-    type: 'highpass',
-    gain: 25
-  }
-};
-
-var beep_fsharp = {
-  type: 'triangle',
-  frequency: 739.99,
-  duration: .2,
-  filter: {
-    frequency: 1000,
-    type: 'highpass',
-    gain: 25
-  }
-};
-
-
-
-
-/* 
- *  Declare namespace
- */
-
-var Synth = function(){
-  this.context = new AudioContext();
-};
-
-
-
-/* 
- *  Accepts a "sound" object and plays it
- */
-
-Synth.prototype.play = function(sound){
-
-  var self = this,
-      osc = self.context.createOscillator(),
-      connection;
-      
-      // Supported: ['sine', 'triangle', 'sawtooth', 'square']
-      osc.type = sound.type;
-      osc.frequency.setValueAtTime(sound.frequency, self.context.currentTime);
-      connection = osc;
-
-      // Manipulate frequency over time
-      if(sound.ramp){
-        osc.frequency.linearRampToValueAtTime(
-          sound.ramp.to, 
-          self.context.currentTime + sound.duration
-        );  
+(function (root) {
+
+   function Synth() {
+     this.context = new AudioContext();
+     this.modulators = [];
+   }
+
+  /*
+   * Some presets - static properties on Synth
+   */
+   Synth.presets = {
+    kick: {
+      type: 'sine',
+      frequency: 100,
+      duration: .3,
+      ramp: {
+        to: 70
       }
-
-      // Apply filter if requested
-      if(sound.filter){
-        var filter = self.context.createBiquadFilter();
-
-        filter.type = sound.filter.type;
-        filter.frequency.value = sound.filter.frequency;
-        filter.gain.value = sound.filter.gain;
-
-        connection.connect(filter);
-        connection = filter;
-
+    },
+    snare: {
+      type: 'square',
+      frequency: 100,
+      duration: .05,
+    },
+    beep_a: {
+      type: 'triangle',
+      frequency: 440,
+      duration: .2,
+      filter: {
+        frequency: 1000,
+        type: 'highpass',
+        gain: 25
       }
-      
-
-      connection.connect(this.context.destination);
-      osc.start(this.context.currentTime);
```

Comment:
@tdoherty the reason why I included the `osc.start` and `osc.stop` here was to enable the duration of "sound" objects. I'd really like to keep it this way because I think keeping the ability to play sounds as a one shot makes it much more accessible for a casual user.

I think your implementation may be better moved into methods of it's own - for example: `Synth.prototype.start` and `Synth.prototype.stop`. 

That would allow for my ultimate vision of a Fruity Loops esque beat designer with one-shots, and `start` and `stop` for virtual instruments.

Thoughts?